Observer/TescoPaymentObserver.php

```diff
@@ -39,19 +39,15 @@ public function execute(Observer $observer)
     {
         $order   = $observer->getEvent()->getOrder();
         $payment = $order->getPayment();
-        
-        if (!$payment) {
```

Comment:
I don't expect that in this callback after order is compleated $payment object will be ever null